src/main/java/com/aws/greengrass/secretmanager/FileSecretDao.java

```diff
@@ -45,26 +47,59 @@ public FileSecretDao(KernelClient kernelClient) throws SecretManagerException {
 
     /**
      * Retrieve secret document from underlying file store.
+     *
      * @return {@link SecretDocument} containing all persisted secrets
      * @throws SecretManagerException when there is any issue reading the store.
      */
     public synchronized SecretDocument getAll() throws SecretManagerException {
+        SecretDocument secretDocFromServiceTopic = getSecretDocFromServiceTopic();
+        SecretDocument secretDocFromRuntimeTopic = getSecretDocFromRuntimeTopic();
```

Comment:
let's prefer data in the runtime. If we have runtime info, use that and only that. If there's nothing there, then fallback to the old one.

Reply:
I think we should check service info irrespective of the runtime info. 

If customer has an older version of the component which wrote the secrets to service topic, then upgraded it to newer version, which writes to both service and runtime topics. Now if the component is downgraded, it will find new secrets in the runtime info but may miss the older ones if it is just used as a fallback (because we have not written the older ones to the runtime config anywhere).

(Also, I'll have to make this a set instead of an array)

Reply:
My thinking is that we want to get rid of the service config at some point because it will be blown away on every single deployment, including during the rollback.

So actually, writing to the service config shouldn't be done anyway. The config will not survive a downgrade. Reading the service config is also pointless because the deployment to upgrade to this version will again blow away the service config.

So actually, I think we can just do the simple thing and simply change where we're reading/writing without any fallback at all.

Reply:
Thinking through how it'd actually be used, it is pointless to try and read or write to the old config. Either an upgrade or a downgrade would wipe out the config anyway; it is guaranteed to be missing. So why bother at all